src/go/cmd/review.go

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"slices"
+	"sort"
+	"strings"
+)
+
+var errExternalModule = errors.New("reviewed module exports a type defined in a different repository")
+
+// Review represents an apiview review of an Azure SDK for Go module
+type Review struct {
+	// modules maps module paths to Modules implicated in this API review. It
+	// contains only the reviewed module in most cases, however it contains
+	// more when the reviewed module exports types defined in another module.
+	modules map[string]*Module
+	// name of the APIView review e.g. "sdk/azcore"
+	name string
+	// path on disk to the reviewed module
+	path string
+	// reviewed is the module being reviewed
+	reviewed *Module
```

Comment:
Does it make sense to store this pointer twice? Its also in the map.

Reply:
We need either this pointer or a key for the map because iteration order over maps is undefined. The choice is pretty arbitrary but the pointer is more to the point and smaller.